salt/search/elasticsearch.sls

```diff
@@ -0,0 +1,82 @@
+# Adapted from:
+# - https://github.com/elifesciences/builder-base-formula/blob/master/elife/elasticsearch.sls
+
+elasticsearch-repo:
+    pkgrepo.managed:
+        - humanname: Official Elasticsearch PPA
+        - name: deb https://artifacts.elastic.co/packages/5.x/apt stable main
+        - dist: stable
+        - file: /etc/apt/sources.list.d/elasticsearch.list
+        - key_url: https://artifacts.elastic.co/GPG-KEY-elasticsearch
+
+elasticsearch:
+    group:
+        - present
+
+    user:
+        - present
+        - groups:
+            - elasticsearch
+        - require:
+            - group: elasticsearch
+
+    pkg:
+        - installed
+        - require:
+            - pkg: oracle-java8-installer
+            - pkgrepo: elasticsearch-repo
+
+    service:
+        - running
+        - enable: True
+        - require:
+            - pkg: elasticsearch
+            - file: /etc/elasticsearch/elasticsearch.yml
+            - group: elasticsearch
+
+elasticsearch-config:
+    file.managed:
+        - name: /etc/elasticsearch/elasticsearch.yml
+        - source: salt://search/config/etc-elasticsearch-elasticsearch.yml
+        - user: elasticsearch
+        - group: elasticsearch
+        - mode: 644
+        - template: jinja
+
+# The GUI is now a stand-alone component and not a plugin/
+# This has increased the complexity of install (last attempt below).
+# For now, unless there are strong objections, I've shelved the GUI.
```

Comment:
Was part of the adapted file here: https://github.com/elifesciences/builder-base-formula/blob/master/elife/elasticsearch.sls

If its not required, we can set up a sense installation for the vagrant machines